agent/agent_test.go

```diff
@@ -967,6 +970,135 @@ func TestAgent_AddServiceWithH2CPINGCheck(t *testing.T) {
 	requireCheckExists(t, a, "test-h2cping-check")
 }
 
+// testTCPServer is a simple TCP echo server for use during tests.
+type testTCPServer struct {
+	l                        net.Listener
+	stopped                  int32
+	accepted, closed, active int32
+}
+
+// newTestTCPServer opens as a listening socket on the given address and returns
+// a TestTCPServer serving requests to it. The server is already started and can
+// be stopped by calling Close().
+func newTestTCPServer(t *testing.T) *testTCPServer {
+	// Yes I get it, we're passing the same cert from the Consul Agent as is
+	// configured in this toy application server. Don't ever do this except in
+	// testing. I see you, thinking about it, don't.
+	certFile := "../test/key/ourdomain_server.cer"
+	keyFile := "../test/key/ourdomain_server.key"
+	caFile := "../test/ca/root.cer"
+	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
+	require.Equal(t, err, nil)
```

Comment:
Here and elsewhere, we should use `require.NoError()`.

```suggestion
	require.NoError(t, err)
```